Invalid CSRF Token

### Description

I run jellyseerr as a docker container behind traefik reverse proxy with SSL. Since updating to 2.4.0, the app is inaccessible with the error `{ "message": "invalid csrf token" }`.

Rolling back to 2.3.0, or disabling CSRF protection, resolves the issue.

### Version

2.4.0

### Steps to Reproduce

See above

### Screenshots

_No response_

### Logs

```shell

```

### Platform

desktop

### Database

SQLite (default)

### Device

Any

### Operating System

Any

### Browser

Any

### Additional Context

_No response_

### Code of Conduct

- [x] I agree to follow Jellyseerr's Code of Conduct

Darn, I had the same issue and preview-csrf-protection-fix didn't work for me.
Back to 2.3.0 for now.

```
jellyseerr  | > jellyseerr@0.1.0 start /app
jellyseerr  | > NODE_ENV=production node dist/index.js
jellyseerr  | 
jellyseerr  | 2025-03-10T23:07:39.330Z [info]: Commit Tag: 71db5842732253b5fb13adda13e578044736806c 
jellyseerr  | node:internal/modules/cjs/loader:1228
jellyseerr  |   throw err;
jellyseerr  |   ^
jellyseerr  | 
jellyseerr  | Error: Cannot find module 'csurf'
jellyseerr  | Require stack:
jellyseerr  | - /app/dist/index.js
jellyseerr  |     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
jellyseerr  |     at Function._load (node:internal/modules/cjs/loader:1055:27)
jellyseerr  |     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
jellyseerr  |     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
jellyseerr  |     at Module.require (node:internal/modules/cjs/loader:1311:12)
jellyseerr  |     at require (node:internal/modules/helpers:136:16)
jellyseerr  |     at Object.<anonymous> (/app/dist/index.js:59:33)
jellyseerr  |     at Module._compile (node:internal/modules/cjs/loader:1554:14)
jellyseerr  |     at Object..js (node:internal/modules/cjs/loader:1706:10)
jellyseerr  |     at Module.load (node:internal/modules/cjs/loader:1289:32) {
jellyseerr  |   code: 'MODULE_NOT_FOUND',
jellyseerr  |   requireStack: [ '/app/dist/index.js' ]
jellyseerr  | }
jellyseerr  | 
jellyseerr  | Node.js v22.14.0
jellyseerr  |  ELIFECYCLE  Command failed with exit code 1.
jellyseerr exited with code 1
```

> Darn, I had the same issue and preview-csrf-protection-fix didn't work for me. Back to 2.3.0 for now.
> 
> ```
> jellyseerr  | > jellyseerr@0.1.0 start /app
> jellyseerr  | > NODE_ENV=production node dist/index.js
> jellyseerr  | 
> jellyseerr  | 2025-03-10T23:07:39.330Z [info]: Commit Tag: 71db5842732253b5fb13adda13e578044736806c 
> jellyseerr  | node:internal/modules/cjs/loader:1228
> jellyseerr  |   throw err;
> jellyseerr  |   ^
> jellyseerr  | 
> jellyseerr  | Error: Cannot find module 'csurf'
> jellyseerr  | Require stack:
> jellyseerr  | - /app/dist/index.js
> jellyseerr  |     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
> jellyseerr  |     at Function._load (node:internal/modules/cjs/loader:1055:27)
> jellyseerr  |     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
> jellyseerr  |     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
> jellyseerr  |     at Module.require (node:internal/modules/cjs/loader:1311:12)
> jellyseerr  |     at require (node:internal/modules/helpers:136:16)
> jellyseerr  |     at Object.<anonymous> (/app/dist/index.js:59:33)
> jellyseerr  |     at Module._compile (node:internal/modules/cjs/loader:1554:14)
> jellyseerr  |     at Object..js (node:internal/modules/cjs/loader:1706:10)
> jellyseerr  |     at Module.load (node:internal/modules/cjs/loader:1289:32) {
> jellyseerr  |   code: 'MODULE_NOT_FOUND',
> jellyseerr  |   requireStack: [ '/app/dist/index.js' ]
> jellyseerr  | }
> jellyseerr  | 
> jellyseerr  | Node.js v22.14.0
> jellyseerr  |  ELIFECYCLE  Command failed with exit code 1.
> jellyseerr exited with code 1
> ```

Because you'll need to repull the image. This was fixed just now

--- a/package.json
+++ b/package.json
@@ -32,6 +32,7 @@
   },
   "license": "MIT",
   "dependencies": {
+    "@dr.pogodin/csurf": "^1.14.1",
     "@formatjs/intl-displaynames": "6.2.6",
     "@formatjs/intl-locale": "3.1.1",
     "@formatjs/intl-pluralrules": "5.1.10",
@@ -51,7 +52,6 @@
     "copy-to-clipboard": "3.3.3",
     "country-flag-icons": "1.5.5",
     "cronstrue": "2.23.0",
-    "csrf-csrf": "^3.1.0",
     "date-fns": "2.29.3",
     "dayjs": "1.11.7",
     "email-templates": "12.0.1",

--- a/server/index.ts
+++ b/server/index.ts
@@ -28,7 +28,7 @@ import restartFlag from '@server/utils/restartFlag';
 import { getClientIp } from '@supercharge/request-ip';
 import { TypeormStore } from 'connect-typeorm/out';
 import cookieParser from 'cookie-parser';
-import { doubleCsrf } from 'csrf-csrf';
+import csurf from 'csurf';
 import type { NextFunction, Request, Response } from 'express';
 import express from 'express';
 import * as OpenApiValidator from 'express-openapi-validator';
@@ -162,23 +162,18 @@ app
       }
     });
     if (settings.network.csrfProtection) {
-      const { doubleCsrfProtection, generateToken } = doubleCsrf({
-        getSecret: () => settings.clientId,
-        cookieName: 'XSRF-TOKEN',
-        cookieOptions: {
-          httpOnly: true,
-          sameSite: 'strict',
-          secure: !dev,
-        },
-        size: 64,
-        ignoredMethods: ['GET', 'HEAD', 'OPTIONS'],
-      });
-
-      server.use(doubleCsrfProtection);
-
+      server.use(
+        csurf({
+          cookie: {
+            httpOnly: true,
+            sameSite: true,
+            secure: !dev,
+          },
+        })
+      );
       server.use((req, res, next) => {
-        res.cookie('XSRF-TOKEN', generateToken(req, res), {
-          sameSite: 'strict',
+        res.cookie('XSRF-TOKEN', req.csrfToken(), {
+          sameSite: true,
           secure: !dev,
         });
         next();

--- a/server/index.ts
+++ b/server/index.ts
@@ -1,3 +1,4 @@
+import csurf from '@dr.pogodin/csurf';
 import PlexAPI from '@server/api/plexapi';
 import dataSource, { getRepository, isPgsql } from '@server/datasource';
 import DiscoverSlider from '@server/entity/DiscoverSlider';
@@ -28,7 +29,6 @@ import restartFlag from '@server/utils/restartFlag';
 import { getClientIp } from '@supercharge/request-ip';
 import { TypeormStore } from 'connect-typeorm/out';
 import cookieParser from 'cookie-parser';
-import csurf from 'csurf';
 import type { NextFunction, Request, Response } from 'express';
 import express from 'express';
 import * as OpenApiValidator from 'express-openapi-validator';

--- a/server/types/custom.d.ts
+++ b/server/types/custom.d.ts
@@ -1,0 +1,4 @@
+declare module '@dr.pogodin/csurf' {
+  import csrf = require('csurf');
+  export = csrf;
+}

--- a/src/utils/fetchOverride.ts
+++ b/src/utils/fetchOverride.ts
@@ -31,7 +31,7 @@ if (typeof window !== 'undefined') {
 
     const headers = {
       ...(init?.headers || {}),
-      ...(csrfToken ? { 'X-CSRF-TOKEN': csrfToken } : {}),
+      ...(csrfToken ? { 'XSRF-TOKEN': csrfToken } : {}),
     };
 
     const newInit: RequestInit = {

